test/others/criu-ns/run.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python
+
+import os
+import shutil
+import signal
+import subprocess
+import sys
+
+CR_NS = "../../.././scripts/criu-ns"
+
+os.chdir(os.getcwd())
+
+
+def check_dumdir():
+    if os.access("dumpdir", os.X_OK):
+        shutil.rmtree("dumpdir")
+        os.mkdir("dumpdir", 0o755)
+
+
+check_dumdir()
+
+# dump with --shell-job
```

Comment:
This is not a universal approach, because you rely on the fact that  test script is run from the shell session. But it may not be true in CI environment. I can suggest to take a look on `shell-job` test https://github.com/checkpoint-restore/criu/blob/criu-dev/test/others/shell-job/run.py

Reply:
Thanks for the review. Can you please let me know what's mean by an "universal approach". Yeah, I have seen the `shell-job` test implementations. It has a few modules (`termios, fcntl, pty`) I'm not familiar at all :) So, I'd really appreciate if you could let me know their use cases or any reading materials regarding them.

Reply:
Thanks for the prompt reply. Understood. But, I think, I do need to test the `--shell-job` [option](https://github.com/checkpoint-restore/criu/issues/1909#issue-1257945578) with regards to `criu-ns` script. So, in this scenario I think I can follow this (current) approach (may be not :) ) And other cases: without --shell-job and --restore-detached, manually allocating TTYs make sense. Right :) ?

Reply:
Let's ask Pavel @Snorch :) He is original author of task

Reply:
First thing:

> We need to test criu-ns
>
>    with --shell-job and without it

I already wrote everything in issue description, we need both cases in test.

Second, cases in test and how test is run in CI are two different things, we should assume no tty given when running in CI, so we should manually create it as it is needed to test shell-job option.